### ISSUE TITLE
wanderer prefs endian issue on be systems.

### ISSUE THREAD

**Issue by wawatok** [AmigaPorts/AROS-mirror#9]
fix endiannes when handling wanderer prefs on big endian systems. needs probably to add endian conversion macros.

**Comment by hth313** [comment]
In `AROS/workbench/prefs/wanderer/wpeditor.c` there is a macro `SAVEVIEWSETTINGSTAG` which saves in little endian it seems using `AROS_LONG2LE`. There are two other writes to `_viewSettings_TagList` (around line 2555 & 2565) with similar code missing the `AROS_LONG2LE` macro.

I do not know if this the correct or only place, but that looked suspicious.

**Comment by wawatok** [comment]
yes. i know this. the thing is that as far as i remember it is actually correct that prefs file remains little endian for all platforms, kalamatee has his say here, the thing is that these fields (after saving) aint read correctly neither by wanderer itself nor by prefs program (if run again it doesnt note any changes). so it looks a bit like the problem is still while saving. i have tried to look at it a good while ago and i dont remember all details, i would have to recheck, which i couldnt since we still have one unresolved issue on real hardware im trying to get a proper disassembly of offending code for toni..
 

Gesendet: Montag, 26. November 2018 um 19:19 Uhr
Von: "Håkan Thörngren" <notifications@github.com>
An: AmigaPorts/AROS-mirror <AROS-mirror@noreply.github.com>
Cc: wawatok <w.tokarski@gmx.net>, Author <author@noreply.github.com>
Betreff: Re: [AmigaPorts/AROS-mirror] wanderer prefs endian issue on be systems. (#9)


In AROS/workbench/prefs/wanderer/wpeditor.c there is a macro SAVEVIEWSETTINGSTAG which saves in little endian it seems using AROS_LONG2LE. There are two other writes to _viewSettings_TagList (around line 2555 & 2565) with similar code missing the AROS_LONG2LE macro.

I do not know if this the correct or only place, but that looked suspicious.

**Comment by hth313** [comment]
Unfortunately I have no setup to test it and the description is a bit vague to outsiders. A bug could have a description (or link) how you can observe the incorrect behaviour. Now it is more like a personal to-do note. It would not have helped in this case anyway, as I do not have a proper setup to test it.

I just noticed that the code is inconsistent here, some writes of tags apply the macro, some don't. It felt like an obvious place to start, though it may not be the real or only issue related to this.

When writing binary like this it becomes endian specific code, though it does not matter which order is used, only that it is consistent. You can choose to always use LE, or just use the endian on the platform at hand (if you are really sure you will never move such files around).

**Comment by wawatok** [comment]
>Unfortunately I have no setup to test it and the description is a bit vague to outsiders. A bug could have a description (or >link) how you can observe the incorrect behaviour. Now it is more like a personal to-do note. It would not have helped in this >case anyway, as I do not have a proper setup to test it.

therefore i said i need to review it to deliver a proper description. what i can say offhand is that it affects wanderer/appearance/advanced options.

>I just noticed that the code is inconsistent here, some writes of tags apply the macro, some don't.

i think not all these fields need to be endian adjusted.

>It felt like an obvious >place to start,

yeah i know.

>When writing binary like this it becomes endian specific code, though it does not matter which order is used, only that it is >consistent. You can choose to always use LE, or just use the endian on the platform at hand (if you are really sure you will >never move such files around).
cross platform consistency is a requirement according to kalamatee, if it wasnt id have already fixed it.

**Comment by wawatok** [comment]
as example, one of the settings that dont take any effect on m68k (big endian) is ICON_TEXTMODE_OUTLINE (wpeditor.c line 2582), it uses SAVEVIEWSETTINGSTAG which is defined in 2282:
`
#define SAVEVIEWSETTINGSTAG(tag, defvalue)                                                                                                  \
    _viewSettings_TagList[_viewSettings_TagCount].ti_Tag = AROS_LONG2LE(tag);                                                               \
    _viewSettings_TagList[_viewSettings_TagCount].ti_Data = AROS_LONG2LE(GetTag32Data(tag, defvalue, _viewSettings_Node->wpedbo_Options));  \
    _viewSettings_TagCount += 1;
`
which apparently is performing wndian conversion while saving (AROS_LONG2LE macro should convert an endian specific long value to platform agnostic little endian)

so it seems correct. i wonder where is the flaw here.

**Comment by wawatok** [comment]
> There are two other writes to `_viewSettings_TagList` (around line 2555 & 2565) with similar code missing the `AROS_LONG2LE` macro
i think these settings are not where the problem is.
